crates/uv-requirements/src/specification.rs

```diff
@@ -223,15 +227,84 @@ impl RequirementsSpecification {
         requirements: &[RequirementsSource],
         constraints: &[RequirementsSource],
         overrides: &[RequirementsSource],
+        groups: &[PipGroupName],
         client_builder: &BaseClientBuilder<'_>,
     ) -> Result<Self> {
         let mut spec = Self::default();
 
+        // Resolve sources into specifications so we know their `source_tree`s
+        let mut requirement_sources = Vec::new();
+        for source in requirements {
+            let source = Self::from_source(source, client_builder).await?;
+            requirement_sources.push(source);
+        }
+
+        // pip `--group` flags specify their own sources, which we need to process here
+        if !groups.is_empty() {
+            // Deduplicate with `BTreeMap` to get deterministic behaviour
+            let mut groups_by_path = BTreeMap::<PathBuf, Vec<GroupName>>::new();
+            for group in groups {
+                groups_by_path
+                    .entry(group.path.clone())
+                    .or_default()
+                    .push(group.name.clone());
```

Comment:
Any concern about these additional hidden clones?

Reply:
None, it happens only once per `uv` invocation and it's like... 1-10 strings realistically.

Reply:
My qualm is more that they're hidden from the caller than anything else, but 👍